handler/src/test/java/io/netty/handler/ssl/JdkSslEngineTest.java

```diff
@@ -141,75 +138,88 @@ final void activate(JdkSslEngineTest instance) {
     private static final String FALLBACK_APPLICATION_LEVEL_PROTOCOL = "my-protocol-http1_1";
     private static final String APPLICATION_LEVEL_PROTOCOL_NOT_COMPATIBLE = "my-protocol-FOO";
 
-    @Parameterized.Parameters(name = "{index}: providerType = {0}, bufferType = {1}, combo = {2}, delegate = {3}")
-    public static Collection<Object[]> data() {
-        List<Object[]> params = new ArrayList<Object[]>();
-        for (ProviderType providerType : ProviderType.values()) {
-            for (BufferType bufferType : BufferType.values()) {
-                params.add(new Object[]{ providerType, bufferType, ProtocolCipherCombo.tlsv12(), true });
-                params.add(new Object[]{ providerType, bufferType, ProtocolCipherCombo.tlsv12(), false });
-
-                if (SslProvider.isTlsv13Supported(SslProvider.JDK)) {
-                    params.add(new Object[] { providerType, bufferType, ProtocolCipherCombo.tlsv13(), true });
-                    params.add(new Object[] { providerType, bufferType, ProtocolCipherCombo.tlsv13(), false });
-                }
+    private Provider provider;
+
+    public JdkSslEngineTest() {
+        super(SslProvider.isTlsv13Supported(SslProvider.JDK));
+    }
+
+    List<JdkSSLEngineTestParam> newJdkParams() {
```

Comment:
It doesn't work if this overrides `newTestParams`?

Reply:
Well `newJdkParams` are only used by the tests that take `ProviderType` into account, which are only a subset of the tests.